tests/test.h

```diff
@@ -83,6 +83,20 @@ void add_unordered_string_output(char* output);
  */
 void check_inputs(uint64_t expected_buttons, float expected_joysticks[4], robot_desc_field_t source);
 
+/******************** UDP Device Data Checks ******************/
+
+/**
+ * Checks that the given device was returned by the UDP thread.
+ * 
+ * Args:
+ *  dev_data: Protobuf struct that is outputted by the UDP thread from get_next_device_data()
+ *  index: index where device should be
```

Comment:
How does a test know which index a particular device would be in the protobuf?

Reply:
They probably just need to run it out on their own using the CLI. it's also the same order as device indices in SHM, but with any gaps removed.